Realm.Shared/PreserveAttribute.cs

```diff
@@ -15,16 +15,27 @@
 // limitations under the License.
 //
 ////////////////////////////////////////////////////////////////////////////
- 
+
+using System;
+
 namespace Realms
 {
     // Tell the linker to preserve a class or method even if it looks like it's not invoked.
     // Since it matches by name, it works to just declare it here.
-    [System.AttributeUsage(System.AttributeTargets.All)]
-    public class PreserveAttribute : System.Attribute
+    [AttributeUsage(System.AttributeTargets.All)]
+    public sealed class PreserveAttribute : Attribute
     {
-        public PreserveAttribute (string reason = null) {}
-        public bool Conditional { get; set; }
+
+        public bool AllMembers;
+        public bool Conditional;
+
+        public PreserveAttribute(string reason = null)
+        {
+        }
+
+        public PreserveAttribute(Type type)
```

Comment:
I actually copied this from the `Foundation.PreserveAttribute` signature. I'll remove the changes here once I've rebased #674 onto this branch.